.github/workflows/bicep_az_deployment_group.yml

```diff
@@ -0,0 +1,232 @@
+name: Perform action for an Az Deployment Group via Bicep
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: What environment is being deployed
+        required: true
+        type: string
+      resourceGroupName:
+        required: true
+        type: string
+      action:
+        description: What action to perform - validate, what-if or create
+        required: true
+        type: string
+    secrets:
+      AZURE_CREDENTIALS:
```

Comment:
It's a bit annoying having to list all the secrets repeatedly, but it should make adding the other environments or having different workflows much easier.